### PR TITLE
Add option to query serverless OpenSearch instances

### PR DESCRIPTION
**Pull request by svennergr**
**What is this feature?**
This PR adds the possibility to also query OpenSearch Serverless services.

**Special notes for your reviewer**:
PR in Grafana to add the needed changes: https://github.com/grafana/grafana/pull/60344

### CODE REVIEW

File: pkg/opensearch/client/client.go [hunk 1]
```diff
@@ -313,6 +319,14 @@ func (c *baseClientImpl) executeRequest(method, uriPath, uriQuery string, body [
 		req.SetBasicAuth(c.ds.User, password)
 	}
 
+	
+	if req.Method != http.MethodGet && c.getSettings().Get("serverless").MustBool(false) {
+		h := sha256.New()
```
Comment (gabor) [inline pkg/opensearch/client/client.go:324]: what do you think about using `sha256.Sum256` ( https://pkg.go.dev/crypto/sha256#Sum256) here? seems shorter.

**Review by gabor — APPROVED** [review]
nice work! i left a small comment.

**Comment by cgowthaman** [comment]
@svennergr version selection is not applicable for OpenSearch serverless. Can we hide the version option on serverless selection

File: pkg/opensearch/client/client.go (hunk 1 above)
Comment (svennergr) [reply pkg/opensearch/client/client.go]: fixed in 4be05df

**Comment by svennergr** [comment]
> @svennergr version selection is not applicable for OpenSearch serverless. Can we hide the version option on serverless selection

Thanks, fixed in a1bfd1a.

**Pull request merged**